google-cloud-bigquerystorage/src/test/java/com/google/cloud/bigquery/storage/v1alpha2/JsonToProtoConverterTest.java

```diff
@@ -0,0 +1,105 @@
+import com.google.protobuf.Descriptors.Descriptor;
+import com.google.protobuf.Descriptors.FieldDescriptor;
+import java.util.Map;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.junit.runners.JUnit4;
+
+@RunWith(JUnit4.class)
+public class JsonToProtoConverterTest {
+  private static ImmutableMap<Table.TableFieldSchema.Type, Descriptor>
+      BQTableTypeToProtoDescriptor =
+          new ImmutableMap.Builder<Table.TableFieldSchema.Type, Descriptor>()
+              .put(Table.TableFieldSchema.Type.BOOL, BoolType.getDescriptor())
+              .put(Table.TableFieldSchema.Type.BYTES, BytesType.getDescriptor())
+              .put(Table.TableFieldSchema.Type.DATE, Int64Type.getDescriptor())
+              .put(Table.TableFieldSchema.Type.DATETIME, Int64Type.getDescriptor())
+              .put(Table.TableFieldSchema.Type.DOUBLE, DoubleType.getDescriptor())
+              .put(Table.TableFieldSchema.Type.GEOGRAPHY, BytesType.getDescriptor())
+              .put(Table.TableFieldSchema.Type.INT64, Int64Type.getDescriptor())
+              .put(Table.TableFieldSchema.Type.NUMERIC, BytesType.getDescriptor())
+              .put(Table.TableFieldSchema.Type.STRING, StringType.getDescriptor())
+              .put(Table.TableFieldSchema.Type.TIME, Int64Type.getDescriptor())
+              .put(Table.TableFieldSchema.Type.TIMESTAMP, Int64Type.getDescriptor())
+              .build();
+
+  private boolean isDescriptorEqual(Descriptor convertedProto, Descriptor originalProto) {
+    for (FieldDescriptor convertedField : convertedProto.getFields()) {
+      FieldDescriptor originalField = originalProto.findFieldByName(convertedField.getName());
+      if (originalField == null) {
+        return false;
+      }
+      FieldDescriptor.Type convertedType = convertedField.getType();
+      FieldDescriptor.Type originalType = originalField.getType();
+      if (convertedType != originalType) {
+        return false;
+      }
+      if (convertedType == FieldDescriptor.Type.MESSAGE) {
+        if (!isDescriptorEqual(convertedField.getMessageType(), originalField.getMessageType())) {
+          return false;
+        }
+      }
+    }
+    return true;
+  }
+
+  @Test
+  public void testBQTableSchemaToProtoDescriptorSimpleTypes() throws Exception {
+    for (Map.Entry<Table.TableFieldSchema.Type, Descriptor> entry :
+        BQTableTypeToProtoDescriptor.entrySet()) {
+      Table.TableFieldSchema tableFieldSchema =
+          Table.TableFieldSchema.newBuilder()
+              .setType(entry.getKey())
+              .setMode(Table.TableFieldSchema.Mode.NULLABLE)
+              .setName("test_field_type")
+              .build();
+      Table.TableSchema tableSchema =
+          Table.TableSchema.newBuilder().addFields(0, tableFieldSchema).build();
+      Descriptor descriptor = JsonToProtoConverter.BQTableSchemaToProtoSchema(tableSchema);
+      assertTrue(isDescriptorEqual(descriptor, entry.getValue()));
```

Comment:
The test error will be hard to debug, it is better to put asserts directly at point of failure.